k9mail/src/main/java/com/fsck/k9/provider/AttachmentProvider.java

```diff
@@ -155,9 +161,14 @@ private String getType(String accountUuid, String id, String mimeType) {
         return type;
     }
 
+    @Nullable
     private ParcelFileDescriptor openAttachment(String accountUuid, String attachmentId) {
         try {
             InputStream inputStream = getAttachmentInputStream(accountUuid, attachmentId);
+            if (inputStream == null) {
```

Comment:
I really don't like using MessagingException all over the place because it's so unspecific and you never know what is actually going on. FileNotFoundException might be appropriate, but throwing it all the way from FileProvider seems like basing the decision on what should be an implementation detail. I think I'm good with a `null` return value for "nothing to open here" still